### PR TITLE
Swathi Fix resources dot centering on Tasks pages

### PR DESCRIPTION
**Pull request by Lemonnycodes**
# Description
This PR addresses the issue of misaligned initials inside resource dots and improves the layout when there are multiple resources in the task table. The initials inside the resource circles are now correctly centered both vertically and horizontally.

## Related PRS (if any):
This frontend PR is related to the development backend PR.

## Main changes explained:
- Updated the CSS file to center the initials in the resource dots using display: flex, justify-content: center, and align-items: center.
- Applied text-align: center and vertical-align: middle to table cells to align text and icons centrally.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to Other Links>Projects>Click WBS Icon>Choose WBS>Edit
6. Verify that the initials in resource circles are properly centered.
7. Add multiple resources (5 or more) to a task and verify that they display correctly without overlap and can be horizontally scrolled if necessary.
8. Verify this feature works in dark mode

## Screenshots or videos of changes:
Before : 
image
After : 
full view
10+ resources
image

### CODE REVIEW

**Review by lucy-xfl — APPROVED** [review]
flex works well in dark and light mode.
WeChatdc4e596b417774654a777a7d030b4fa9
ssets/dee352cd-a9e0-432d-ba64-61e2b315ede4)

**Review by nikhilpittala16 — DISMISSED** [review]
I have tested the PR and everything works as expected.

https://github.com/user-attachments/assets/e3fe6a21-37f2-44ba-a720-477f2876723a

**Review by SharadhaS24 — DISMISSED** [review]
Validated the addition of multiple resource types (1, 5, or more) in both light and dark modes, ensuring consistent performance and correct functionality. Its centered and aligned properly.

image

image

image

**Comment by Anoushka21** [comment]
Tested in both modes. Aligned and working well.

https://github.com/user-attachments/assets/792c0971-52fe-402a-90d7-3b3e3b675df8

**Review by Anoushka21 — APPROVED** [review]

**Comment by pallavithorat** [comment]
I have verified the PR by ensuring that the initials in the resource circles are properly centered, and multiple resources display correctly with horizontal scrolling, without overlap. The feature also works seamlessly in dark mode.
PR#2789

**Review by NK3101 — APPROVED** [review]
Tested the PR in both light and dark modes. Working as expected.

**Review by shashankhv — APPROVED** [review]
Verified initials centering, resource display, and tested in dark mode. Works as expected.
[swathi-PR.webm](https://github.com/user-attachments/assets/a3fa334a-a236-42cb-acb4-93badbd479ef)

**Review by CodewithKoushica — CHANGES_REQUESTED** [review]
The changes in this PR work well, and the initials in the resource dots are correctly centered. However, I noticed that the UI makes a noticeable jump when expanding and collapsing the resource section. It would improve the user experience if this transition could be made smoother, possibly by adding a CSS transition effect. Other than that, the fix looks great!


https://github.com/user-attachments/assets/28a3c155-2a2a-436d-bbe6-7e817b263db2

**Review by nikhilgiri0226 — APPROVED** [review]
I have tested the PR. Everything works as expected. The resources icons with initials were aligned properly inside the column horizontally. I have tested it in different screen sizes.


https://github.com/user-attachments/assets/f24c12f7-c0ba-4939-bd26-2d16b9b0c92b

**Review by zhiminV — DISMISSED** [review]
I have tested that the initials in resource circles are properly centered.
and 5 or more resources display correctly without overlap and can be horizontally. And I also verify this feature works in dark mode.
Screenshot 2024-10-25 at 11 25 35 PM
Screenshot 2024-10-25 at 11 20 19 PM

Screenshot 2024-10-25 at 11 26 16 PM

**Review by Anoushka21 — APPROVED** [review]

**Review by varun-elango — DISMISSED** [review]
The code changes look good and the functionality is working as expected. The initials are properly centered and multiple resources are displayed correctly without any overlap. Verified in dark mode as well.
PR2789_1
PR2789_2
PR2789_3

**Review by bhavya17prakash — APPROVED** [review]
Working as intended in dark mode as well
PR 2789

**Review by Rupa2400 — APPROVED** [review]
Tested the PR in both light and dark modes. The PR is working as expected.

PR_2789_1

PR_2789_2

PR_2789_3

File: src/components/Projects/WBS/WBSDetail/wbs.css [hunk 1]
```diff
@@ -120,9 +123,20 @@
...
   padding: 4px 8px;
   line-height: 1.2;
-  height: 40px
+  height: 40px;
+  text-align: center;
+  vertical-align: middle;
+}
+.table th {
+  text-align: center; /* Centers the text horizontally */
+  vertical-align: middle; /* Centers the text vertically */
```
Comment (sheetalmangate) [inline src/components/Projects/WBS/WBSDetail/wbs.css:132]: Comments for CSS classes are not required. they are self explanatory

**Comment by sheetalmangate** [comment]
CSS changes working as per requirement stated. UI changes looks good, just having small suggestion
1)  comments for CSS classes are not required they are self explanatory.

Approving this PR

PR2789_testing

**Review by sheetalmangate — APPROVED** [review]

**Review by sheetalmangate — DISMISSED** [review]

**Review by samarth9201 — DISMISSED** [review]
Reviewed the PR Working as expected.
image

**Review by anirudhsk2107 — DISMISSED** [review]
Changes are working in light mode and dark mode

Screen Shot 2024-10-29 at 10 36 55 PM
Screen Shot 2024-10-29 at 10 36 47 PM

**Review by Rui-Liu-github — APPROVED** [review]
I reviewed PR #2789 by pulling the latest version from Swathi. The fixed CSS styling works as expected, with the resources dot positioned at the center.

image
image

**Review by SupriyaSudini — APPROVED** [review]
I have verified that the initials in the resource circles are properly centered and tested adding multiple resources (5 or more) to a task to ensure they display correctly without overlap, with horizontal scrolling available when needed.

image

image

image

**Review by AshritaCherlapally — APPROVED** [review]
This code is working as expected, when creating task with adding resources, the resources are aligned on center as expected

https://github.com/user-attachments/assets/8163e20e-655a-4580-bec0-daaad8d49ac8

**Review by audreydieuanh — DISMISSED** [review]
It works correctly both in light and dark mode and with 5+ resources.
Screenshot 2024-11-02 at 9 17 55 PM
Screenshot 2024-11-02 at 9 18 03 PM

**Review by hqzhu0913 — COMMENTED** [review]
The resources tags center as expected. 
But I find a problem with limited resources selections. I am not sure if I didn't test it correctly. But these are not problem of your UI change. Comment me if you see the problem as well.
Screenshot 2024-11-05 at 1 31 37 PM


Screenshot 2024-11-05 at 1 31 05 PM

**Comment by humera314** [comment]
I ran the tests on this feature, and it’s working perfectly across all specified conditions, including dark mode.
image
image

**Review by SS-Reddys — APPROVED** [review]
Changes are working as expected

image

image

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**